### PR TITLE
Add documentation for Multipart form and URL encoded in SDK 

### PR DESCRIPTION
**Pull request by gabrielsim1993**

### CODE REVIEW

**Comment by gabrielsim1993** [comment]
@eeshan92

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Form URL Encoded
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: should be **Form urlencoded**

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+This request is declared in the execute portion of the connector actions framework. This is an additional request option that is appended to the back of the base request.
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: This request format can be declared in any blocks (`execute`, `acquire`, `fields` etc.) in your custom adapter code. It should be chained to one of the [base request](/developing-connectors/sdk/walk-through.html#base-request).

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+## Example
+HubSpot has an endpoint to send form submission data to HubSpot. This [endpoint](https://developers.hubspot.com/docs/methods/forms/submit_form) accepts a form urlencoded request.
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: Let's use the **Submit data to a form** endpoint in [HubSpot API](https://developers.hubspot.com/docs/methods/forms/submit_form) as an example. This endpoint accepts form data in form urlencoded format.

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          { name: "pageUrl", type: :string },
+          { name: "pageName", type: :string },
+          { name: "redirectUrl", type: :string }
+        ]
+      end,
+
+      execute: lambda do |connection, input|
+        post("https://forms.hubspot.com/uploads/form/v2/#{{input['portal_id']}}/#{{input['form_guid']}}").
+          request_format_www_form_urlencoded.
+          request_body(
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: you can just pass the entire input hash (rejecting `portal_id` and `form_guid`)

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+curl command:
+```sh
+curl -X POST \
+  -H 'Content-Type: application/x-www-form-urlencoded' \
+  -d 'firstname=TestContact&lastname=FormSub&email=formsub@hubspot.com&newcustomproperty=testing&hs_context=%7B%22hutk%22%3A%2260c2ccdfe4892f0fa0593940b12c11aa%22%2C%22ipAddress%22%3A%22192.168.1.12%22%2C%22pageUrl%22%3A%22http%3A%2F%2Fdemo.hubapi.com%2Fcontact%2F%22%2C%22pageName%22%3A%22Contact%2BUs%22%2C%22redirectUrl%22%3A%22http%3A%2F%2Fdemo.hubapi.com%2Fthank-you%2F%22%7D'
+  https://forms.hubspot.com/uploads/form/v2/12345/67890 \
+```
+
+Workato:
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: More details here is required. Break down the request into each part and explain where each method match the cURL example.

File: developing-connectors/sdk/data-format/form-url-encoded.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+# Form URL Encoded
+
+This request is declared in the execute portion of the connector actions framework. This is an additional request option that is appended to the back of the base request.
+
+## Example
+HubSpot has an endpoint to send form submission data to HubSpot. This [endpoint](https://developers.hubspot.com/docs/methods/forms/submit_form) accepts a form urlencoded request.
+
+curl command:
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/form-url-encoded.md]: A cURL example looks like this.

File: developing-connectors/sdk/data-format/request_format_multipart_form.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
+# Multipart form
+
+[Multipart form request](https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4.2)  is typically used to send large files and data to a server.
+
+This request is declared in the execute portion of the connector actions framework. This is an additional request option that is appended to the back of the base request.
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/request_format_multipart_form.md]: This request format can be declared in any blocks (`execute`, `acquire`, `fields` etc.) in your custom adapter code. It should be chained to one of the base request.

File: developing-connectors/sdk/data-format/request_format_multipart_form.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+## Example
+IBM Watson has an endpoint to convert a document. This [endpoint](https://www.ibm.com/watson/developercloud/document-conversion/api/v1/#convert-document) accepts a multipart/form-data request.
+
+curl command:
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/request_format_multipart_form.md]: A cURL example looks like this.

File: developing-connectors/sdk/data-format/request_format_multipart_form.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+```sh
+curl \
+  -X POST \
+  -u "{username}":"{password}" \
+  -F config="{\"conversion_target\":\"answer_units\"}" \
+  -F "file=@sample.pdf;type=application/pdf" \
+  "https://gateway.watsonplatform.net/document-conversion/api/v1/convert_document?version=2015-12-15"
+```
+
+Workato:
```
Comment (eeshan92) [inline developing-connectors/sdk/data-format/request_format_multipart_form.md]: Same thing here. More details needed. Breakdown each part and point to the cURL equivalent.

Comment (eeshan92) [reply developing-connectors/sdk/data-format/request_format_multipart_form.md]: Most importantly, explain how using an array of length 2 is the way to declare form data, as compared to a regular key/value pair.

**Comment by gabrielsim1993** [comment]
@eeshansim changes made

**Comment by gabrielsim1993** [comment]
@eeshansim switched the syntax

**Pull request merged**